### PR TITLE
codegen: Fix dictionary handling and semantics

### LINKED ISSUE #11152: mozbrowsererror detail.type is undefined
It's supposed to be "fatal". 

I believe it has to do with `type_` here: https://github.com/servo/servo/blob/e2a5b1cb710b4c5d19d0493e2a304152c42feabe/components/script/dom/htmliframeelement.rs#L323

/cc @asajeffrey

### PR DESCRIPTION
**Pull request by emilio**
Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data:
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #11152 (github issue number if applicable).

Either:
- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. 

Fixes #11152

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11155)

### CODE REVIEW

**Comment by emilio** [comment]
r? @asajeffrey 

Not sure about how to test it, since we only trigger mozbrowser errors on panics... I'll look for some other reserved keyword in the username.

**Comment by asajeffrey** [comment]
Indeed, testing is going to be tricky.

**Comment by asajeffrey** [comment]
r=me once the testing story is worked out.

**Comment by emilio** [comment]
I'm having local problems with the test I've just written, so I'm going to do a try run.

The problem is that for some reason even when the `isKeywordTypeSuccess` function in the bindings output is called, the rust function `IsKeywordTypeSuccess` isn't, and the script thread goes into an idle state before finishing to execute the script.

So the only output I'm seeing is:

```
created
received: 
success
```

I'm fearing this is a bug in our conversions for `TestDictionary` that somehow corrupts the stack, but I'm not sure.

Running this with `debug-mozjs` gives no clues, and the same result.

**Comment by emilio** [comment]
@bors-servo: try- clean try

(Fixed tidy)

**Comment by jdm** [comment]
Rather than corrupting the stack, I suspect that a generated binding function is returning false at some point which silently suspends JS execution.

**Comment by emilio** [comment]
This should be ready for re-review.

The problem was that when we convert a rust dictionary to js, we setted the properties even when they were `None` (that is the value we use to represent that a prop isn't setted in JS). It seems this  was not the correct behaviour per http://heycam.github.io/webidl/#es-dictionary, and the PR changes it.

Going to do a try run to check if we rely on the old semantics:

@bors-servo: try

**Comment by emilio** [comment]
@bors-servo: retry

 * #11165 

More than probably a new intermittent, passes locally.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 1]
```diff
@@ -638,8 +638,11 @@ def getJSToNativeConversionInfo(type, descriptorProvider, failureCode=None,
     # Note that we can't just set failureCode to exceptionCode, because setting
     # failureCode will prevent pending exceptions from being set in cases when
     # they really should be!
+    unhandledExceptionCode = exceptionCode
     if exceptionCode is None:
-        exceptionCode = "return false;"
+        exceptionCode = 'return false;'
+        unhandledExceptionCode = 'debug!("Unhandled exception error"); return false;'
```
Comment (Ms2ger) [inline components/script/dom/bindings/codegen/CodegenRust.py]: I'm not sure what this is about; you're returning `false`, so it is handled.

Comment (emilio) [inline components/script/dom/bindings/codegen/CodegenRust.py]: The point is that we don't set any exception and the script will abort silently... I think it's worth to print a message in this case, but if you think it's not I can remove it.

Comment (emilio) [inline components/script/dom/bindings/codegen/CodegenRust.py]: (Abort in the sense of stop the execution of the script, not in the sense of `SIGABRT`).

**Comment by emilio** [comment]
Updated test expectations, the cors test seems to be intermittent, the other failures are old expected timeouts uncovered by this patch.

@bors-servo: r=Ms2ger

**Comment by emilio** [comment]
@bors-servo: retry

 * https://github.com/servo/servo/issues/11100

**Pull request merged**